File: mobile/androidApp/src/main/kotlin/com/mentora/android/ui/components/CourseProgressCard.kt
package com.mentora.android.ui.components

import androidx.compose.ui.Modifier
import androidx.compose.runtime.Composable
import androidx.compose.ui.res.stringResource
import com.mentora.android.R
import kotlin.math.roundToInt

/**
 * `design-system/COMPONENTS.md` § CourseProgressCard (lines 305-307): "same shell as CourseCard —
 * always shows `ProgressBar` + '% complete' + 'Resume' `TonalButton`, and drops rating/student-count
 * row." Delegates to the same internal [BaseCourseCard] as [CourseCard] (shared shell, per that
 * file's own kdoc), always passing `progress`/`progressLabel`, never a `metaRow`.
 */
@Composable
fun CourseProgressCard(
    title: String,
    instructorName: String,
    seed: String,
    categoryId: String?,
    categoryLabel: String,
    thumbnailContentDescription: String,
    progress: Float,
    onResumeClick: () -> Unit,
    modifier: Modifier = Modifier,
    mediaId: String? = null,
    thumbnailUrl: String? = null,
    onClick: (() -> Unit)? = null,
    // T19 review fix (LOW-1, D94): generalizing ErrorState.retryLabel's own HIGH fix — see that
    // component's kdoc. `progressLabelFormatter`'s own English literal is untouched here — it's a
    // pre-existing, already-disclosed gap (every real call site overrides it; no localized formatter
    // exists yet), not part of this fix's scope.
    resumeLabel: String = stringResource(R.string.course_progress_card_resume_label),
    progressLabelFormatter: (Float) -> String = { p -> "${(p * 100).roundToInt()}% complete" },
) {
    BaseCourseCard(
        title = title,
        instructorName = instructorName,
        seed = seed,
        categoryId = categoryId,
        categoryLabel = categoryLabel,
        mediaId = mediaId,
        thumbnailUrl = thumbnailUrl,
        thumbnailContentDescription = thumbnailContentDescription,
        actionLabel = resumeLabel,
        onActionClick = onResumeClick,
        modifier = modifier,
        onClick = onClick,
        metaRow = null, // dropped per spec.
        progress = progress,
        progressLabel = progressLabelFormatter(progress),
    )
}

File: mobile/iosApp/iosApp/Components/CourseProgressCard.swift
import SwiftUI
import shared

// Phase 5 Task T11 slice 2 (Component Kit B) -- `design-system/COMPONENTS.md § CourseProgressCard`
// (lines 305-307): "same shell as CourseCard -- always shows ProgressBar + '% complete' + 'Resume'
// TonalButton, and drops the rating/student-count row." Delegates to the same `BaseCourseCard` as
// `CourseCard.swift` (shared shell, per that file's own header), always passing `progress`/
// `progressLabel`, never a `metaRow`. Ported directly from Android's own `CourseProgressCard.kt`.
struct CourseProgressCard: View {
    let title: String
    let instructorName: String
    let seed: String
    let categoryId: String?
    let categoryLabel: String
    let thumbnailAccessibilityLabel: String
    let progress: Double
    let onResumeClick: () -> Void
    var mediaId: String? = nil
    var thumbnailUrl: String? = nil
    var onClick: (() -> Void)? = nil
    /// DISCLOSED DEVIATION from Android's own default-parameter convenience: Android's
    /// `resumeLabel: String = stringResource(R.string.course_progress_card_resume_label)` can resolve a
    /// default inline because a `@Composable` always has an ambient `Context`/locale to call
    /// `stringResource` with. `MentoraStrings.text(_:locale:)` (this key: the real, pre-existing
    /// `course_progress_card_resume_label`, confirmed present in `Resources/Localizable.xcstrings`
    /// before this task began -- zero new keys added) instead requires an explicit `AppLocale`, and this
    /// file (a plain `Components/` composite, matching every atom/composite in this kit's own "never
    /// resolves `MentoraStrings` itself" convention) has no `AppEnvironment` of its own to source one
    /// from -- so `resumeLabel` is REQUIRED here, always resolved and passed in by the caller.
    var resumeLabel: String
    var progressLabelFormatter: (Double) -> String = { p in "\(Int((p * 100).rounded()))% complete" }

    var body: some View {
        BaseCourseCard(
            title: title,
            instructorName: instructorName,
            seed: seed,
            categoryId: categoryId,
            categoryLabel: categoryLabel,
            mediaId: mediaId,
            thumbnailUrl: thumbnailUrl,
            thumbnailAccessibilityLabel: thumbnailAccessibilityLabel,
            actionLabel: resumeLabel,
            onActionClick: onResumeClick,
            onClick: onClick,
            metaRow: nil, // dropped per spec.
            progress: progress,
            progressLabel: progressLabelFormatter(progress)
        )
    }
}
